doc_helper.js

```diff
@@ -0,0 +1,304 @@
+    
+    // Use a more general regex to capture all event emissions
+    // This captures the event name and whatever is passed as the second argument
+    const regex = /svc_event\.emit\(['"]([^'"]+)['"]\s*,\s*([^)]+)\)/g;
+    let match;
+    
+    while ((match = regex.exec(content)) !== null) {
+        const eventName = match[1];
+        const eventId = `core.${eventName}`;
+        const eventArg = match[2].trim();
+        
+        // Check if this file contains code that might affect event.allow
+        const hasAllowEffect = content.includes('event.allow') || 
+                              content.includes('.allow =') || 
+                              content.includes('.allow=');
+        
+        // Skip if we've already seen this event
+        if (seenEvents.has(eventId)) continue;
+        
+        // Generate description based on event name
+        let description = generateDescription(eventName);
+        let propertyDetails = {};
+        
+        // Case 1: Inline object - extract properties directly
+        if (eventArg.startsWith('{')) {
+            // Extract properties from inline object
+            const propertiesMatch = eventArg.match(/{([^}]*)}/);
+            if (propertiesMatch) {
+                const propertiesText = propertiesMatch[1];
+                extractProperties(propertiesText, propertyDetails, hasAllowEffect, eventName);
+            }
+        } 
+        // Case 2: Variable reference - find variable definition
+        else {
+            const varName = eventArg.trim();
+            // Look for variable definition patterns like: const event = { prop1: value1 };
+            const varDefRegex = new RegExp(`(?:const|let|var)\\s+${varName}\\s*=\\s*{([^}]*)}`, 'g');
+            let varMatch;
+            
+            if ((varMatch = varDefRegex.exec(content)) !== null) {
+                const propertiesText = varMatch[1];
+                extractProperties(propertiesText, propertyDetails, hasAllowEffect, eventName);
+            }
+        }
+        
+        // Add the event to our collection
+        seenEvents.set(eventId, {
+            id: eventId,
+            event: eventName,
+            filename: path.basename(filePath),
+            description: description,
+            properties: propertyDetails
+        });
+    }
+}
+
+// Helper function to extract properties from a properties text string
+function extractProperties(propertiesText, propertyDetails, hasAllowEffect, eventName) {
+    const properties = propertiesText
+        .split(/\s*,\s*/)
+        .map(prop => prop.split(':')[0].trim())
+        .filter(prop => prop);
+    
+    // Generate property details
+    properties.forEach(prop => {
+        propertyDetails[prop] = {
+            type: guessType(prop),
+            mutability: hasAllowEffect ? 'effect' : 'no-effect',
+            summary: guessSummary(prop, eventName)
+        };
+    });
+}
+
+function generateDescription(eventName) {
```

Comment:
I really like this approach. We need a way to override these in case the generated description is wrong. (also consider this for other generated text)